src/test/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticStack.java

```diff
@@ -58,70 +46,65 @@ public class ElasticStack extends DockerComposeContainer<ElasticStack> {
     private ElasticBackend elasticBackendConfiguration;
     private LogStorageRetriever elasticsearchRetriever;
 
+    private static final String suffixService(String service) {
+        return service + SERVICE_SUFFIX;
+    }
+
     public ElasticStack() {
         super(new File("src/test/resources/docker-compose.yml"));
         withExposedService(ELASTICSEARCH_SERVICE, ELASTICSEARCH_PORT)
                 .withExposedService(KIBANA_SERVICE, KIBANA_PORT)
                 .withExposedService(EDOT_SERVICE, OTEL_PORT)
+                .waitingFor(ELASTICSEARCH_SERVICE, new DockerHealthcheckWaitStrategy())
+                .waitingFor(KIBANA_SERVICE, new DockerHealthcheckWaitStrategy())
                 .waitingFor(EDOT_SERVICE, new DockerHealthcheckWaitStrategy())
                 .withStartupTimeout(Duration.ofMinutes(10));
     }
 
-    /**
-     * @return the RestClientBuilder to create the Elasticsearch REST client.
-     */
-    public Rest5ClientBuilder getBuilder() {
-        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(USER_NAME, PASSWORD.toCharArray());
-
-        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-        try {
-            credentialsProvider.setCredentials(new AuthScope(HttpHost.create(getEsUrl())), credentials);
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException(e);
-        }
-        CloseableHttpAsyncClient httpclient = HttpAsyncClients.custom()
-                .setDefaultCredentialsProvider(credentialsProvider)
-                .build();
-
-        Rest5ClientBuilder builder = Rest5Client.builder(URI.create(getEsUrl())).setHttpClient(httpclient);
-        return builder;
-    }
-
     /**
      * @return The URL to access to the Elasticsearch Docker container.
      */
     public String getEsUrl() {
-        return "http://" + this.getServiceHost(ELASTICSEARCH_SERVICE, ELASTICSEARCH_PORT) + ":"
-                + this.getServicePort(ELASTICSEARCH_SERVICE, ELASTICSEARCH_PORT);
+        return "http://localhost:" + this.getServicePort(ELASTICSEARCH_SERVICE, ELASTICSEARCH_PORT);
```

Comment:
Do not use hardcoded URLs, use the service name to get the host, the correct service name.